backend/src/main/java/reviewme/review/service/ReviewGatheredLookupService.java

```diff
@@ -0,0 +1,56 @@
+package reviewme.review.service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import reviewme.question.domain.Question;
+import reviewme.question.repository.QuestionRepository;
+import reviewme.review.domain.Answer;
+import reviewme.review.repository.AnswerRepository;
+import reviewme.review.service.dto.response.gathered.ReviewsGatheredBySectionResponse;
+import reviewme.review.service.exception.ReviewGroupNotFoundByReviewRequestCodeException;
+import reviewme.review.service.exception.SectionNotFoundInTemplateException;
+import reviewme.review.service.mapper.ReviewGatherMapper;
+import reviewme.reviewgroup.domain.ReviewGroup;
+import reviewme.reviewgroup.repository.ReviewGroupRepository;
+import reviewme.template.domain.Section;
+import reviewme.template.repository.SectionRepository;
+
+@Service
+@RequiredArgsConstructor
+public class ReviewGatheredLookupService {
+
+    private final QuestionRepository questionRepository;
+    private final AnswerRepository answerRepository;
+    private final ReviewGroupRepository reviewGroupRepository;
+    private final SectionRepository sectionRepository;
+
+    private final ReviewGatherMapper reviewGatherMapper;
+
+    @Transactional(readOnly = true)
+    public ReviewsGatheredBySectionResponse getReceivedReviewsBySectionId(String reviewRequestCode, long sectionId) {
+        ReviewGroup reviewGroup = reviewGroupRepository.findByReviewRequestCode(reviewRequestCode)
+                .orElseThrow(() -> new ReviewGroupNotFoundByReviewRequestCodeException(reviewRequestCode));
+        Section section = sectionRepository.findByIdAndTemplateId(sectionId, reviewGroup.getTemplateId())
+                .orElseThrow(() -> new SectionNotFoundInTemplateException(sectionId, reviewGroup.getTemplateId()));
+
+        Map<Long, Question> questionIdQuestion = questionRepository
+                .findAllBySectionIdOrderByPosition(section.getId()).stream()
+                .collect(Collectors.toMap(Question::getId, Function.identity()));
+        List<Answer> receivedAnswers = answerRepository.findReceivedAnswersByQuestionIds(
+                reviewGroup.getId(), new ArrayList<>(questionIdQuestion.keySet()));
+        Map<Long, List<Answer>> questionIdAnswers = receivedAnswers.stream()
+                .collect(Collectors.groupingBy(Answer::getQuestionId));
+
+        Map<Question, List<Answer>> questionAnswers = questionIdQuestion.entrySet()
+                .stream()
+                .collect(Collectors.toMap(Map.Entry::getValue, entry -> questionIdAnswers.getOrDefault(entry.getKey(), List.of())));
```

Comment:
테드가 말한 흐름과 비슷하게 이렇게 로직만 있어도 될 것 같아요!
1. 섹션에 대한 모든 질문을 가져오기
2. 리뷰 그룹에 해당하는 답변만 가져오기
3. 두개 겹치는 걸로 매핑하기
(questionRepository 찌르는 건 캐싱할 거니까 괜찮다고 생각해서요~)
```suggestion
        List<Long> questionIdsBySection = questionRepository.findIdsBySectionIdOrderByPosition(section.getId());
        Map<Long, List<Answer>> answersByQuestionId = answerRepository.findByReviewGroupId(reviewGroup.getId())
                .stream()
                .collect(Collectors.groupingBy(Answer::getQuestionId));
        
        Map<Question, List<Answer>> questionAnswers = questionIdsBySection.stream()
                .collect(Collectors.toMap(
                        questionId -> questionRepository.findById(questionId).orElseThrow(),
                        questionId -> answersByQuestionId.getOrDefault(questionId, new ArrayList<>())
                ));
```

Reply:
반영하긴 했는데..! 의도한게 맞으신지 한번 봐주세용🥹

Reply:
맞아요!! 💯💯💯